### ISSUE TITLE
QCarousel 2.0

### ISSUE THREAD

**Issue by hawkeye64** [quasarframework/rfcs#4]
# RFC (Request for Comments) Template

## Please select your type of RFC (one or more):
_Put an `x` in the boxes that apply_

- [X] Change ?
- [ ] Method ?
- [ ] Behavior ?
- [ ] Research ?
- [X] Innovation ?

## Other

- Start Date: 2020-02-13
- Target Major Version: 2.x
- Reference Issues: 
https://github.com/quasarframework/quasar/issues/6135
https://github.com/quasarframework/quasar/issues/5924
https://github.com/quasarframework/quasar/issues/5702
https://github.com/quasarframework/quasar/issues/5412
https://github.com/quasarframework/quasar/issues/4714
https://github.com/quasarframework/quasar/issues/1670

## Does this introduce a breaking change?
_Put an `x` in the box that applies_

- [X] Yes
- [ ] No

## Proposed RFC
One of the most talked about and asked for features.

Examples
https://swiperjs.com/
https://kenwheeler.github.io/slick/
https://flickity.metafizzy.co/

**Comment by N1Mirzo** [comment]
Hi, @hawkeye64 What about an e-commerce carousel.
Please add an e-commerce carousel too.
A carousel component that has product cards for instance 4 cards in a carousel.
They are responsive. This means the cards become 3 when the carousel is in the PC tablet screen size. In larger phone devices they left 2. And on the normal screen phones it left one.
And the cards will not be slided. Only one slide will be slided. And if you add opportunity to slide ony one card. It will be useful too. But I am asking you to add something that's responsive.

**Comment by N1Mirzo** [comment]
![image](https://user-images.githubusercontent.com/55438076/74606820-4864da00-50f5-11ea-9e62-8e13544b2d1f.png)
This is on mobile phone.
Just look that's worse. Here the problem is how to cutting one card from the first slide and adding it to second slide.

**Comment by m0jimo** [comment]
1) I'd find useful to have a property for not covering the carousel content by horizontal/vertical (thumbnails or radio buttons).

Examples: 
![Snímek obrazovky 2020-04-08 v 18 07 08](https://user-images.githubusercontent.com/37825539/78807026-c8a40f00-79c3-11ea-880b-b0fabac85481.png)

![Snímek obrazovky 2020-04-08 v 18 06 17](https://user-images.githubusercontent.com/37825539/78806947-aca06d80-79c3-11ea-838e-367cd75109a5.png)

Example of navigation with position out of the carousel content: 
![Snímek obrazovky 2020-04-08 v 18 09 44](https://user-images.githubusercontent.com/37825539/78807303-26385b80-79c4-11ea-8035-c6e4ac810a1d.png)

2) color property for non-active content instead of manually replacing it (current non-active color is close to original and to easy to distinguished from active one).
```
 .q-carousel__navigation-icon--inactive {
    opacity: 0.5 !important;
    color: grey !important;
  }
```

**Comment by hawkeye64** [comment]
@m0jimo You can change placement by a number of ways. look at this:
![image](https://user-images.githubusercontent.com/10262924/78942266-8daee380-7a76-11ea-897d-33f87609e6a1.png)
This is done by adding `no-margin` to your `q-carousel-slide` items
```html
        <q-carousel-slide
          v-for="featured in mutatedFeatured"
          :key="featured.id"
          :name="featured.slug"
          class="column flex-center no-margin"
        >

```

**Comment by m0jimo** [comment]
Thanks for the tip. Tried to find a live example of your screenshot but with no success. I also tried recommended `no-margin` change and it doesn't change the position in my code and the same result I have on [Codepen](https://codepen.io/m0jimo/pen/mdeePvd). I tried thumbnails and navigation. 
But I'm sure it can be done by rewriting/extending styles of `q-carusel__navigation ...` in the similar way as for navigation colors.

**Comment by hawkeye64** [comment]
@m0jimo Check it out. I changed background color so you can see the navigation. You can change the color though, so you could still have a white background if you like: https://codepen.io/Hawkeye64/pen/eYpZpKr

**Comment by ghost** [comment]
> ![image](https://user-images.githubusercontent.com/55438076/74606820-4864da00-50f5-11ea-9e62-8e13544b2d1f.png)
> This is on mobile phone.
> Just look that's worse. Here the problem is how to cutting one card from the first slide and adding it to second slide.

You are right, it looks horrible on the phone, i think we need something like slick carousel

**Comment by ImSeaWorld** [comment]
I made a component that's a bit rough around the edges, but I think having specificity with a breakpoints or options prop would be necessary. I made a quick and dirty approach that takes in a string `xs:1,0;sm:2,0;md:3;lg:4;3840:5`

Explaination - `{breakpoint|screen width}:{perSlide},{resetIndex};`

When the component is made it'll convert any breakpoints(ex `xs`, `sm` etc) to their min screen width equivalent(if it exists). Then watch the screen width(`$q.screen.width`) cross checking the breakpoints stored. When one is found, the data is split based on the `perSlide` value given; then reset the index to the provided `resetIndex` or `1` by default. This keeps from the component being out of sync when resize events take place. They're entirely optional.

https://codepen.io/imseaworld/pen/mdaVGWp

This isn't my first implementation of it, but certainly the most dynamic that I can plop into different projects. I've used this for reviews to gallery preview. This is certainly needed for edge cases. I hope this helps towards the RFC

**Comment by geoidesic** [comment]
@ImSeaWorld I don't know if it's just me but the codepen link doesn't seem to do anything other than show some code as white text in grey boxes? [UPDATE: never mind... I see one has to resize the window to make it do something]

**Comment by ImSeaWorld** [comment]
> @ImSeaWorld I don't know if it's just me but the codepen link doesn't seem to do anything other than show some code as white text in grey boxes? [UPDATE: never mind... I see one has to resize the window to make it do something]

I updated it to make it easier to see what's happening. Apologies for the bad UX 😄

![chrome_PF7utGd2GY](https://github.com/quasarframework/rfcs/assets/20188588/cd9cc5c8-977a-4373-a24e-996c7e79382d)
